crates/task-impls/src/consensus/proposal_helpers.rs

```diff
@@ -873,7 +894,15 @@ pub async fn handle_quorum_proposal_validated<TYPES: NodeType, I: NodeImplementa
                                 "Updating consensus state with decided upgrade certificate: {:?}",
                                 cert
                             );
-                            task_state.decided_upgrade_cert = Some(cert.clone());
+                            #[cfg(not(feature = "dependency-tasks"))]
+                            {
+                                task_state.decided_upgrade_cert = Some(cert.clone());
+                            }
+
+                            #[cfg(feature = "dependency-tasks")]
+                            {
+                                decided_upgrade_cert = Some(cert.clone());
+                            }
```

Comment:
If I understand correctly what's happening here, I think you can move `decided_upgrade_cert` out of `ConsensusTaskState` and just have it in `Consensus`. We take a lock on consensus in this function anyway, I don't see why we can't just store this there (and avoid gating on the feature).

Reply:
So moving `decided_upgrade_cert` out of `Consensus` already exists, this is required due to the `consensus` variable being a read lock, not a write lock, when this callback is invoked. As a result, I need to be able to capture the value. I tried doing it with a write lock, but the `consensus.visit_leaf_anscestors` call is an immutable reference, so the compiler gets angry. The next-best option in this case was to simply have a variable that I allocate and assign.